src/disjoint_mut.rs

```diff
@@ -1212,3 +1214,34 @@ impl<T> Default for DisjointMutArcSlice<T> {
         [].into_iter().collect()
     }
 }
+
+impl<T: FromZeroes> DisjointMutArcSlice<T> {
+    pub fn new_zeroed_slice(len: usize) -> Self {
+        #[cfg(debug_assertions)]
+        let inner = {
+            let box_slice = Box::new_zeroed_slice(len);
+            Arc::new(DisjointMut::new(box_slice))
+        };
+        #[cfg(not(debug_assertions))]
+        let inner = {
+            use std::mem;
+
+            let arc_slice = Arc::<[T]>::new_zeroed_slice(len);
+
+            // Do our best to check that `DisjointMut` is in fact `#[repr(transparent)]`.
+            type A = Vec<u8>; // Some concrete sized type.
+            const _: () = assert!(mem::size_of::<DisjointMut<A>>() == mem::size_of::<A>());
+            const _: () = assert!(mem::align_of::<DisjointMut<A>>() == mem::align_of::<A>());
+
+            // SAFETY: When `#[cfg(not(debug_assertions))]`, `DisjointMut` is `#[repr(transparent)]`,
+            // containing only an `UnsafeCell`, which is also `#[repr(transparent)]`.
+            unsafe { mem::transmute::<Arc<[_]>, Arc<DisjointMut<[_]>>>(arc_slice) }
```

Comment:
Ideally, `Arc` transmutes should be done by going through `Arc::into_raw`→`pointer::cast`→`Arc::from_raw` instead of a transmute, since the `struct Arc` itself is not `#[repr(transparent)]`.

Reply:
Oh yes, that's true, thanks for catching this!

Reply:
* Fixed in #1361.